### ISSUE TITLE
reflection_pad2d passed the performance test, but no performance data was collected. Pending confirmation from the development team.

### ISSUE THREAD

**Issue by lovywu** [flagos-ai/FlagGems#2178]
### Please describe your question

============================= test session starts ==============================
platform linux -- Python 3.12.3, pytest-8.1.1, pluggy-1.6.0
rootdir: /home/week_test_2026/20260325/FlagGems
configfile: pytest.ini
plugins: anyio-4.9.0, hypothesis-6.130.8, flakefinder-1.1.0, rerunfailures-15.1, shard-0.1.2, xdist-3.6.1, xdoctest-1.0.2, typeguard-4.3.0
collected 394 items / 334 deselected / 60 selected
Running 60 items in this shard

test_special_perf.py ................................................... [ 85%]
.........                                                                [100%]

=============================== warnings summary ===============================
../../../../../usr/local/lib/python3.12/dist-packages/triton/runtime/autotuner.py:99: 31 warnings
  /usr/local/lib/python3.12/dist-packages/triton/runtime/autotuner.py:99: DeprecationWarning: warmup, rep, and use_cuda_graph parameters are deprecated. See https://github.com/triton-lang/triton/pull/4496 for details.
    warnings.warn(("warmup, rep, and use_cuda_graph parameters are deprecated. See "

<frozen importlib._bootstrap>:488
  <frozen importlib._bootstrap>:488: DeprecationWarning: builtin type SwigPyPacked has no __module__ attribute

<frozen importlib._bootstrap>:488
  <frozen importlib._bootstrap>:488: DeprecationWarning: builtin type SwigPyObject has no __module__ attribute

<frozen importlib._bootstrap>:488
  <frozen importlib._bootstrap>:488: DeprecationWarning: Type google._upb._message.MessageMapContainer uses PyType_Spec with a metaclass that has custom tp_new. This is deprecated and will no longer be allowed in Python 3.14.

<frozen importlib._bootstrap>:488
  <frozen importlib._bootstrap>:488: DeprecationWarning: Type google._upb._message.ScalarMapContainer uses PyType_Spec with a metaclass that has custom tp_new. This is deprecated and will no longer be allowed in Python 3.14.

../../../../../usr/local/lib/python3.12/dist-packages/google/protobuf/internal/well_known_types.py:93
  /usr/local/lib/python3.12/dist-packages/google/protobuf/internal/well_known_types.py:93: DeprecationWarning: datetime.datetime.utcfromtimestamp() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.fromtimestamp(timestamp, datetime.UTC).
    _EPOCH_DATETIME_NAIVE = datetime.datetime.utcfromtimestamp(0)

benchmark/test_special_perf.py::test_reflection_pad2d_benchmark_tensor[padding0-dtype0-shape0]
  /usr/local/lib/python3.12/dist-packages/torch/library.py:364: UserWarning: Warning only once for all operators,  other operators may also be overridden.
    Overriding a previously registered kernel for the same operator and the same dispatch key
    operator: aten::_flash_attention_forward(Tensor query, Tensor key, Tensor value, Tensor? cum_seq_q, Tensor? cum_seq_k, SymInt max_q, SymInt max_k, float dropout_p, bool is_causal, bool return_debug_mask, *, float? scale=None, SymInt? window_size_left=None, SymInt? window_size_right=None, Tensor? seqused_k=None, Tensor? alibi_slopes=None) -> (Tensor output, Tensor softmax_logsumexp, Tensor rng_state, Tensor unused, Tensor debug_attn_mask)
      registered at /pytorch/build/aten/src/ATen/RegisterSchema.cpp:6
    dispatch key: CUDA
    previous kernel: registered at /pytorch/torch/csrc/autograd/generated/VariableType_0.cpp:18228
         new kernel: registered at /home/week_test_2026/20260325/FlagGems/src/flag_gems/__init__.py:540 (Triggered internally at /pytorch/aten/src/ATen/core/dispatch/OperatorEntry.cpp:218.)
    self.m.impl(

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=============== 60 passed, 334 deselected, 37 warnings in 24.40s ===============

sys:1: DeprecationWarning: builtin type swigvarlink has no __module__ attribute

**Comment by bin913** [comment]
Issue #2178 Fixed in Current Master
Root‑cause Analysis
The test_reflection_pad2d_benchmark_tensor test in the issue shows status "passed but no performance data", for the following reason:
The old implementation (test_reflection_pad2d_benchmark_tensor introduced in commit 5589e3bc4) bypassed the benchmark framework. It directly called triton.testing.do_bench inside the test and only printed output via print(f"... Speedup: {speedup:.2f}x"):
Results were only printed to stdout and never flowed through the framework’s BenchmarkMetrics / emit_record_logger recording pipeline;
Consequently pytest reported PASSED, yet no performance data for this op appeared in aggregated results or log records.
Fix (Already Present in Current Master)
PR #2273 (commit 076edc420, 2026‑04‑08) titled "fix benchmark issues for reflection_pad2d and nll_loss_nd":
Removed test_reflection_pad2d_benchmark_tensor (the print‑only variant);
Rewrote it using standard ReflectionPad2dBenchmark(Benchmark) + get_input_iter + bench.run(), going through the framework recording pipeline;
Also fixed nll_loss_nd_input_fn missing yield inp, target, which caused the same missing‑data symptom.
After subsequent benchmark refactoring (#2661), the repository now maintains a standalone test_reflection_pad2d.py with structure aligned to this fix.
Verification status:
git merge‑base --is‑ancestor 076edc420 upstream/master → the fix is contained in current master;
run() in the framework’s current base.py collects latency_base / latency / speedup / gbps, and persists records via emit_record_logger(result.to_json()).
